sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowExec.scala

```diff
@@ -213,10 +218,24 @@ case class WindowExec(
           (expressions, schema) =>
             newMutableProjection(expressions, schema, subexpressionEliminationEnabled))
 
+        def createOrdering(): Ordering[InternalRow] = {
+          val exprs = orderSpec.map(_.child)
+          val sortExprs = orderSpec.zipWithIndex.map { case (e, i) =>
+            SortOrder(BoundReference(i, e.dataType, e.nullable), e.direction)
+          }
+          val ordering = newOrdering(sortExprs, Nil)
+          new Ordering[InternalRow] {
+            override def compare(x: InternalRow, y: InternalRow): Int = {
+              ordering.compare(newMutableProjection(exprs, child.output)(x),
```

Comment:
I am pretty sure this creates two new mutable projections for each comparison.

Reply:
I should refine this.